### ISSUE TITLE
Duplicate error message (esp. when `pretty: true`)

### ISSUE THREAD

**Issue by kitsonk** [ezolenko/rollup-plugin-typescript2#103]
#### What happens and why it is wrong

Duplicate semantic errors get logged to the console, which causes confusion.  For example:

```
./node_modules/rollup/bin/rollup -c ../../rollup.config.js -i ../../js/main.ts -o gen/bundle/main.js --silent
[!] (rpt2 plugin) Error: ../../js/main.ts:8:1 - error TS2304: Cannot find name 'confounded'.

8 confounded;
  ~~~~~~~~~~

../../js/main.ts
Error: ../../js/main.ts:8:1 - error TS2304: Cannot find name 'confounded'.

8 confounded;
  ~~~~~~~~~~

    at error (/Users/kkelly/github/deno/third_party/node_modules/rollup/dist/rollup.js:3365:15)
    at Object.error (/Users/kkelly/github/deno/third_party/node_modules/rollup/dist/rollup.js:20727:13)
    at RollupContext.error (/Users/kkelly/github/deno/third_party/node_modules/rollup-plugin-typescript2/src/rollupcontext.ts:36:18)
    at lodash_2 (/Users/kkelly/github/deno/third_party/node_modules/rollup-plugin-typescript2/src/print-diagnostics.ts:37:10)
    at arrayEach (/Users/kkelly/github/deno/third_party/node_modules/rollup-plugin-typescript2/node_modules/lodash/lodash.js:516:11)
    at forEach (/Users/kkelly/github/deno/third_party/node_modules/rollup-plugin-typescript2/node_modules/lodash/lodash.js:9342:14)
    at printDiagnostics (/Users/kkelly/github/deno/third_party/node_modules/rollup-plugin-typescript2/src/print-diagnostics.ts:9:2)
    at Object.transform (/Users/kkelly/github/deno/third_party/node_modules/rollup-plugin-typescript2/src/index.ts:238:5)
    at /Users/kkelly/github/deno/third_party/node_modules/rollup/dist/rollup.js:20750:41
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

I am not entirely sure how much this is `rollup-plugin-typescript2` and how much of this is just the way rollup behaves, but I couldn't find an answer either way.

#### Versions
- typescript: 3.0.1
- rollup: 0.63.2
- rollup-plugin-typescript2: 0.16.1

**Comment by vineetit1991** [comment]
any input on this ? I am also facing this issue

**Comment by ezolenko** [comment]
I don't get it with just any error, could you create a test repo with reproduction?

Might depend on error type (typescript only, or something that would trip rollup itself) execution mode (clean build / rebuild / watch) and rollup/plugin settings.

This is what I usually get:
(3 `there were errors or warnings` lines for 3 bundles that are generated in that build)

```
rpt2: there were errors or warnings.
rpt2: rolling caches
rpt2: there were errors or warnings.
rpt2: rolling caches
rpt2: there were errors or warnings.
rpt2: rolling caches
(!) Plugin rpt2: src/index.ts:316:18 - error TS2304: Cannot find name 'Plugin5Context'.

316     _onwrite(this: Plugin5Context, _output: OutputOptions): void
                       ~~~~~~~~~~~~~~

src\index.ts
```

**Comment by agilgur5** [comment]
Going to close this out as a reproduction was unsuccessful and this issue has not received any activity in a few years (and may not be relevant in newer versions anymore).

Can re-open later if warranted on current versions and with a repro.

**Comment by agilgur5** [comment]
So I ended up reproducing this issue while testing #372 .

Per https://github.com/ezolenko/rollup-plugin-typescript2/pull/372#issuecomment-1171953101, the _last_ error will indeed get printed out twice in all cases. It just happens to be a lot more visible when using `tsconfig`'s [`pretty`](https://www.typescriptlang.org/tsconfig#pretty) option as OP is using and as #372 makes the default.

Per https://github.com/ezolenko/rollup-plugin-typescript2/pull/372#issuecomment-1171991058, this duplication actually happens within Rollup though, and not because of this plugin. Rollup [prints out](https://github.com/rollup/rollup/blob/56c1514c7b0d0c985685ed497bdc6ce33b023383/cli/logging.ts#L14) the error message, and then [also prints](https://github.com/rollup/rollup/blob/56c1514c7b0d0c985685ed497bdc6ce33b023383/cli/logging.ts#L31) the error [stack](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack), which can actually contain the error message in it, causing duplication.

This may depend on runtime though as `Error.stack` is non-standardized. It may also depend on how one uses Rollup as well, since this code is in the Rollup CLI specifically.

I'll have a PR out here to _workaround_ this in rpt2 shortly. Will also have to make a similar PR to Rollup itself to handle this upstream. The PR here will be written to be forward-compatible with newer versions of Rollup if that gets merged upstream as well.

**Comment by agilgur5** [comment]
https://github.com/rollup/plugins/issues/376 shows and mentions a "duplicated code frame" for `@rollup/plugin-babel` as well, so this seems to be a long-standing issue that has never quite been fixed.
I couldn't find references to this in Rollup proper, in other issues in `rollup/plugins`, or elsewhere here, so I guess this hasn't been noticed that often, and seemingly only noticed by people looking to get "prettified" output (either in TS or in Babel).

**Comment by agilgur5** [comment]
#373 works around this Rollup bug and has been released in [`0.33.0`](https://github.com/ezolenko/rollup-plugin-typescript2/releases/0.33.0.1)

**Comment by agilgur5** [comment]
This has been fixed upstream in Rollup in https://github.com/rollup/rollup/pull/4749 and released in Rollup [v3.7.5](https://github.com/rollup/rollup/blob/v3.26.2/CHANGELOG.md#375)
